module/zcommon/zpool_prop.c

```diff
@@ -135,6 +138,18 @@ zpool_prop_init(void)
 	    PROP_ONETIME, ZFS_TYPE_POOL, "TNAME");
 	zprop_register_hidden(ZPOOL_PROP_MAXDNODESIZE, "maxdnodesize",
 	    PROP_TYPE_NUMBER, PROP_READONLY, ZFS_TYPE_POOL, "MAXDNODESIZE");
+	zprop_register_hidden(ZPOOL_PROP_SMALLBLKCEILING, "smallblkceiling",
+	    PROP_TYPE_NUMBER, PROP_ONETIME, ZFS_TYPE_POOL, "SMALLBLKCEILING");
+
+	/* hidden boolean properties */
+	zprop_register_index(ZPOOL_PROP_SEGREGATE_LOG,
+	    "segregate_log", 0, PROP_ONETIME, ZFS_TYPE_POOL, "on",
+	    "SEGREGATE_LOG", boolean_table);
+	prop_tbl[ZPOOL_PROP_SEGREGATE_LOG].pd_visible = B_FALSE;
+	zprop_register_index(ZPOOL_PROP_SEGREGATE_SPECIAL,
+	    "segregate_special", 0, PROP_ONETIME, ZFS_TYPE_POOL, "on",
+	    "SEGREGATE_SPECIAL", boolean_table);
+	prop_tbl[ZPOOL_PROP_SEGREGATE_SPECIAL].pd_visible = B_FALSE;
```

Comment:
is there a reason that these manually set  `pd_visible`? Could this be made into a `zpool _register_*()` function